src/Rating.php

```diff
@@ -84,10 +89,10 @@ public function getRates($originPostalCode, $postalCode, $weight, array $options
      * @return array
      *  The list of options with the option-code.
      */
-    private function getOptionCodes(array $options) {
+    protected function parseOptionCodes(array $options) {
       $valid_options= [];
       foreach ($options['option_codes'] as $optionCode) {
-        if (!in_array(strtoupper($optionCode), $this::$VALID_OPTION_CODES)) {
+        if (!in_array(strtoupper($optionCode), self::getOptionCodes())) {
           break;
         }
```

Comment:
**CHANGE REQUEST**

*Logic*

One more change. We need to have consistent logic. Either do not pass the option codes at all if there is at least one that is invalid, or pass all option codes that are valid. Here you are passing the option codes that are valid until the first error.

Just use `continue` instead of `break` and you're good.